src/binder/bindertracing.cpp

```diff
@@ -231,3 +262,123 @@ namespace BinderTracing
         return m_ignoreBind;
     }
 }
+
+namespace BinderTracing
+{
+    ResolutionAttemptedOperation::ResolutionAttemptedOperation(BINDER_SPACE::AssemblyName *assemblyName)
+        : m_pAssemblyName{assemblyName}
+    {
+    }
+
+    ResolutionAttemptedOperation::~ResolutionAttemptedOperation()
+    {
+        TraceCurrentStage();
+    }
+
+    void ResolutionAttemptedOperation::SetResult(HRESULT hr, BINDER_SPACE::Assembly *assembly)
+    {
+        if (SUCCEEDED(hr))
+        {
+            m_result = Result::Success;
+        }
+        else if (hr == HRESULT_FROM_WIN32(ERROR_FILE_NOT_FOUND))
+        {
+            m_result = Result::AssemblyNotFound;
+        }
+        else if (hr == FUSION_E_APP_DOMAIN_LOCKED)
+        {
+            m_result = Result::IncompatibleVersion;
+        }
+        else if (hr == FUSION_E_REF_DEF_MISMATCH)
+        {
+            m_result = Result::MismatchedAssemblyName;
+        }
+        else
+        {
+            m_result = Result::Unknown;
+            UNREACHABLE();
```

Comment:
This implies we don't set the result for really bad failures (anything we don't recognize above) - that is weird. I would expect the tracing to log any failure, regardless if it can decode its meaning or not.

Reply:
Done.